### ISSUE TITLE
Add SoapUI functional tests

### ISSUE THREAD

**Issue by panagiotisbellias** [jcilacad/aeroplanner-rest-api#124]
Create SoapUI project that will test as much functionality as possible through the API endpoints configuring application to use another PostgreSQL database named e.g. 'test'. Also integrate the SoapUI project in the GitHub Actions workflow to run them automated through CLI in each release!

**Comment by panagiotisbellias** [comment]
@jcilacad If you agree you can assign me this issue

**Comment by jcilacad** [comment]
Sure, you can proceed with this idea.

**Comment by panagiotisbellias** [comment]
@jcilacad Can you download/install SoapUI and import the functional tests xml file to tell me your opinion about alternative flows and test suite/test cases structure before proceeding to make the request logic?

Thanks

**Comment by jcilacad** [comment]
@panagiotisbellias Could you please provide me with the location or source from where I can download the functional tests XML file for importing into SoapUI?

**Comment by panagiotisbellias** [comment]
Sure, you can find it [here](https://github.com/panagiotisbellias/osc-aeroplanner-rest-api/blob/issue-124-soapui-tests/Aeroplanner-REST-API-soapui-project.xml)

**Comment by jcilacad** [comment]
thanks

**Comment by panagiotisbellias** [comment]
@jcilacad 
I restructured the tests and seem better now. The only thing left is arranging the flow and run-automate the tests
You can find them [here](https://github.com/panagiotisbellias/osc-aeroplanner-rest-api/blob/issue-124-soapui-tests/AeroplannerREST_API_Functional%20Tests-soapui-project.xml)

**Comment by jcilacad** [comment]
Nice work

**Comment by jcilacad** [comment]
@panagiotisbellias. Do you currently use SoapUI for functional testing in your work? Or would you recommend it as a testing method for the tasks at hand? I’m considering studying it as well.

**Comment by panagiotisbellias** [comment]
> @panagiotisbellias. Do you currently use SoapUI for functional testing in your work? Or would you recommend it as a testing method for the tasks at hand? I’m considering studying it as well.

Yes, I use it in my work for functional, end to end and performace testing. I also recommend it as it automates the testing process

**Comment by jcilacad** [comment]
thanks

**Comment by panagiotisbellias** [comment]
I have added more tests adding test cases about the needed authentication per cases and only the endpoints with id are pending.
I want to inform that I cannot be anymore occupied with this task unless I have time which is hard currently. In my previous commit I have a comment of a possible bug found. It would be a pleasure for more devs or @jcilacad to occupy with this task. Feel free to unassign me this task. In my opinion it can be removed from this release (v1.0.0)

PS: 'Flights Management' TS should be split and placed before and after 'Reservations Management' TS possibly merging the two TSs in 'Flight Reservations Management' TS rearranging TCs

**Comment by jcilacad** [comment]
Thanks for your work; I’ll take it from here and handle the unassignment.
